lib/core/src/display/shape/glyph/system.rs

```diff
@@ -0,0 +1,231 @@
+//! A main glyph system implementation.
+
+use crate::prelude::*;
+
+use crate::display::shape::glyph::font::FontId;
+use crate::display::shape::glyph::font::FontRenderInfo;
+use crate::display::shape::glyph::font::FontRegistry;
+use crate::display::shape::glyph::pen::PenIterator;
+use crate::display::shape::glyph::msdf::MsdfTexture;
+use crate::display::symbol::material::Material;
+use crate::display::symbol::shader::builder::CodeTemplate;
+use crate::display::world::*;
+use crate::system::gpu::data::texture::*;
+use crate::system::gpu::types::*;
+
+use nalgebra::Vector2;
+use nalgebra::Vector4;
+
+
+
+// =============
+// === Glyph ===
+// =============
+
+/// A glyph rendered on screen. The displayed character will be stretched to fit the entire bbox of
+/// underlying sprite.
+#[derive(Shrinkwrap)]
+pub struct Glyph {
+    context         : Context,
+    #[shrinkwrap(main_field)]
+    sprite          : Sprite,
+    msdf_index_attr : Attribute<f32>,
+    color_attr      : Attribute<Vector4<f32>>,
+    font_id         : FontId,
+    msdf_uniform    : Uniform<Texture<GpuOnly,Rgb,u8>>,
+}
+
+impl Glyph {
+    /// Glyph color attribute accessor.
+    pub fn color(&self) -> Attribute<Vector4<f32>> {
+        self.color_attr.clone_ref()
+    }
```

Comment:
Your colors will soon break. Let's leave them this way now, but soon I will be switching to CIE-LAB color mixing in linear space and then your colors will blow up :P Lets talk about it som day - just a note.